proof/courier.go

```diff
@@ -1300,6 +1306,55 @@ func FetchProofProvenance(ctx context.Context, originLocator Locator,
 	// is a reversal of that found in the proof file.
 	var reversedProofs []Blob
 	for {
+		// Before we attempt to fetch the proof from the potentially
+		// remote universe, we'll check our local archive to see if we
+		// already have it. This doesn't make sense in all cases, so we
+		// allow the local archive to be nil, in which case we just skip
+		// this step.
+		var (
+			fullProof Blob
+			err       error
+		)
+		if localArchive != nil {
+			fullProof, err = localArchive.FetchProof(
+				ctx, currentLocator,
+			)
+		}
+
+		// If we don't have a local archive, then the IsFile() check
+		// returns false, and we skip this block.
+		if err == nil && fullProof.IsFile() {
```

Comment:
nit: may be useful to have more comment at the beginning of the block, took a bit to see how this lined up with the explanation in the commit msg.

Reply:
Makes sense, added!